src/kostentraeger/edifact/parser.ts

```diff
@@ -0,0 +1,347 @@
+                    break
+                case "VKG": // Verknüpfung
+                    vkgList.push(readVKG(elements))
+                    break
+                case "NAM": // Name
+                    nam = readNAM(elements)
+                    break
+                case "ANS": // Anschrift
+                    ansList.push(readANS(elements))
+                    break
+                case "ASP": // Ansprechpartner
+                    aspList.push(readASP(elements))
+                    break
+                case "UEM": // Übermittlungsmedium
+                    uemList.push(readUEM(elements))
+                    break
+                case "DFU": // Datenfernübertragung
+                    dfuList.push(readDFU(elements))
+                    break
+            }
+        } catch (error) {
+            error.message = messageTxt + " " + error.message
+            throw error
+        }
+    })
+
+    if (!idk) {
+        throw new Error(`${messageTxt} The mandatory segment "IDK" is missing`)
+    }
+    if (!vdt) {
+        throw new Error(`${messageTxt} The mandatory segment "VDT" is missing`)
+    }
+    if (!nam) {
+        throw new Error(`${messageTxt} The mandatory segment "NAM" is missing`)
+    }
+    if (!fkt) {
+        throw new Error(`${messageTxt} The mandatory segment "FKT" is missing`)
+    }
+    if (ansList.length == 0) {
+        throw new Error(`${messageTxt} At least one "ANS" element is required`)
+    }
+    uemList.forEach((uem) => {
+        if (uem.uebermittlungsmediumSchluessel == "1" && dfuList.length == 0) {
+            throw new Error(`${messageTxt} Data inconsistency: At least one "DFU" element is required if UEM.uebermittlungsmediumSchluessel = 1`)
+        }
+    })
+
+    aspList.sort((a, b) => a.index - b.index)
+    dfuList.sort((a, b) => a.index - b.index)
+
+    return {
+        id: messageId,
+        idk: idk,
+        vdt: vdt,
+        fkt: fkt,
+        nam: nam,
+        kto: kto,
+        vkgList: vkgList,
+        ansList: ansList,
+        aspList: aspList,
+        dfuList: dfuList,
+        uemList: uemList
+    }
+}
+
+const readIDK = (e: string[]): IDK => ({
+    ik: e[0],
+    institutionsart: e[1], 
+    abbreviatedName: e[2],
+    vertragskassennummer: e.length > 3 ? parseInt(e[3]) : undefined
+})
+
+const readVDT = (e: string[]): VDT => ({
+    validityFrom: parseDate(e[0]),
+    validityTo: e[1] ? parseDate(e[1]) : undefined
+})
+
+const readFKT = (e: string[]): FKT => {
+    const k = e[0]
+    if (!verarbeitungskennzeichenSchluessel.hasOwnProperty(k)) {
+        throw new Error(`Unknown VerarbeitungskennzeichenSchluessel "${k}"`)
+    }
+    return { verarbeitungskennzeichenSchluessel: k as VerarbeitungskennzeichenSchluessel }
+}
+
+const readKTO = (e: string[]): KTO => {
+    const accountNumber = e[0] ? e[0] : undefined
+    const bankCode = e[1] ? e[1] : undefined
+    const iban = e[4] ? e[4] : undefined
+    const bic = e[5] ? e[5] : undefined
```

Comment:
I understand the code to always set an undefined value instead of an empty string or another falsy value.
If you also find it easier to write and read, you could write the following to replace an empty string with undefined using `e[0] || undefined`.